solidstart/github-oauth/src/middleware.ts

```diff
@@ -1,40 +1,36 @@
-import { createMiddleware, appendHeader, getCookie, getHeader } from "@solidjs/start/server";
-import { Session, User, verifyRequestOrigin } from "lucia";
+import { createMiddleware} from "@solidjs/start/middleware";
+import { verifyRequestOrigin } from "lucia";
 import { lucia } from "./lib/auth";
+import {appendHeader, getCookie, getHeader} from "vinxi/http";
 
 export default createMiddleware({
-	onRequest: async (event) => {
-		if (event.node.req.method !== "GET") {
-			const originHeader = getHeader(event, "Origin") ?? null;
-			const hostHeader = getHeader(event, "Host") ?? null;
+	onRequest:[ async (event) => {
+		if (event.request.method !== "GET") {
+			const originHeader = getHeader("Origin") ?? null;
+			const hostHeader = getHeader( "Host") ?? null;
 			if (!originHeader || !hostHeader || !verifyRequestOrigin(originHeader, [hostHeader])) {
-				event.node.res.writeHead(403).end();
+
+				event.response.status = 403;
 				return;
 			}
 		}
 
-		const sessionId = getCookie(event, lucia.sessionCookieName) ?? null;
+		const sessionId = getCookie(lucia.sessionCookieName) ?? null;
 		if (!sessionId) {
-			event.context.session = null;
-			event.context.user = null;
+			event.locals.session = null;
+			event.locals.user = null;
 			return;
 		}
 
 		const { session, user } = await lucia.validateSession(sessionId);
 		if (session && session.fresh) {
-			appendHeader(event, "Set-Cookie", lucia.createSessionCookie(session.id).serialize());
+			appendHeader("Set-Cookie", lucia.createSessionCookie(session.id).serialize());
 		}
 		if (!session) {
-			appendHeader(event, "Set-Cookie", lucia.createBlankSessionCookie().serialize());
+			appendHeader("Set-Cookie", lucia.createBlankSessionCookie().serialize());
 		}
-		event.context.session = session;
-		event.context.user = user;
-	}
+		event.locals.session = session;
+		event.locals.user = user;
+	}]
 });
 
-declare module "vinxi/server" {
```

Comment:
How is `context.Locals` typed?

Reply:
In `createMiddleware`, I updated the code to add a type declaration to the event: `FetchEvent` so that the types will be more clear to anyone looking at the code.

To answer your question, `event.locals` is of type `RequestEventLocals`, which is of type: 
```
export interface RequestEventLocals {
    [key: string | symbol]: any;
}
```

Reply:
Is there no way to override the `RequestEventLocals` type?

Reply:
Code added to overide/clarify the `RequestEventLocals` type addition.

(On a different, funny tangential note, I suddenly had a surreal feeling just now where it hit me that reading our back-and-forth on this thread sounds a little like a ChatGPT code interaction :-) Then I thought, "I am not a LLM!", but then I though, "Would anyone know if I were?" Like a reverse Turing test.)(And since text doesn't communicate emotion or intention, I'm laughing as I write this! By the way, this is a great chance for me to thank you for Artic, Oslo, and your re-purposing Lucia as an education opportunity. I really appreciate it, and I thank you. I agree with your decision about Lucia, and it has benefited me. I'm glad that I'm able to give back in this little way of providing some code back to you.).